controller/k8s/api.go

```diff
@@ -414,6 +415,7 @@ func (api *API) GetPodsFor(obj runtime.Object, includeFailed bool) ([]*corev1.Po
 	var namespace string
 	var selector labels.Selector
 	var pods []*corev1.Pod
+	var uid types.UID
```

Comment:
This might be more clear if it was named `ownerUID` or something like that to indicate this is the UID of the owning resource, not the UID of the pod itself.